lib/datadog/tracing/contrib/action_pack/action_dispatch/instrumentation.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require_relative '../../../metadata/ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module ActionPack
+        module ActionDispatch
+          # Instrumentation for ActionDispatch components
+          module Instrumentation
+            module_function
+
+            def set_http_route_tag(http_route)
+              return if http_route.nil?
+
+              active_span = Tracing.active_span
+              return unless active_span
+
+              active_span.set_tag(
+                Tracing::Metadata::Ext::HTTP::TAG_ROUTE,
+                http_route.gsub(/\(.:format\)$/, '')
+              )
+            end
+
+            # Instrumentation for ActionDispatch::Journey components
+            module Journey
+              # Instrumentation for ActionDispatch::Journey::Router
+              # for Rails versions older than 7.1
+              module Router
+                def find_routes(req)
+                  result = super(req)
+
+                  return response unless Tracing.enabled?
+
+                  begin
+                    # Journey::Router#find_routes retuns an array for each matching route.
+                    # This array is [match_data, path_parameters, route].
+                    # We need the route object, since it has a path with route specification.
+                    current_route = result.last&.last&.path&.spec
+                    return result unless current_route
+
+                    # When Rails is serving requests to Rails Engine routes, this function is called
+                    # twice: first time for the route on which the engine is mounted, and second
+                    # time for the internal engine route.
+                    last_route = Tracing.active_span&.get_tag(Tracing::Metadata::Ext::HTTP::TAG_ROUTE)
+                    Instrumentation.set_http_route_tag(last_route.to_s + current_route.to_s)
```

Comment:
Is this concatenating two routes or parts of the same route? If two routes, don't you need some kind of a separator?

Reply:
In my local testing, all the routes had a `/` in the beginning (both rails routes and rails engine routes). I could not find a case where there is no slash at the beginning of the route.

Reply:
These are 2 parts of the same route: one for the route on which the rails engine is mounted, and the second one is the internal rails engine route.

```ruby
  namespace :api do
    mount AuthEngine::Engine => "/auth"
  end
```

In this example we will have 2 parts:  `/api/auth` and `/sign-in(/:expires_at)`.

Reply:
In the codebase rails also [uses a term "route" for those](https://github.com/rails/rails/blob/main/actionpack/lib/action_dispatch/journey/router.rb#L126). I think the problem here is that if you have a rails engine mounted at some route, this `#find_routes` will be called twice: once for the path on which the engine is mounted, and the second time for the internal engine route.

So we only have 2 "routes" here when using rails engines.

What do you think, should I then just keep the term "route" here, or rename it to "route fragment" so that the code is more readable?